### ISSUE TITLE
Error: Dynamic require of \"crypto\" is not supported

### ISSUE THREAD

**Issue by boringContributor** [aws/aws-sam-cli#5824]
I saw this issue being raised multiple times, and despite seeing it should be fixed with version 1.85 I still don't see this working in my setup. Opening this issue to better keep track as the original one is closed.

```
 ApiHandlerFunction:
    Type: AWS::Serverless::Function
    Properties:
      CodeUri: lambda/ApiHandlerFunction/
      Handler: src/index.handler
      Runtime: nodejs18.x
      Architectures:
        - arm64 
    Metadata: # Manage esbuild properties
      BuildMethod: esbuild
      BuildProperties:
        Format: esm
        Minify: true
        Target: "es2020"
        OutExtension:
        - .js=.mjs
        EntryPoints: 
        - src/index.ts 
        Banner:
        - js=import { createRequire } from 'module'; const require = createRequire(import.meta.url);
```

this is my simplified template.yml.
I'm using `sam cli 1.95`

When I deploy I get this error:

```
{
    "errorType": "Error",
    "errorMessage": "Dynamic require of \"crypto\" is not supported",
    "stack": [
        "Error: Dynamic require of \"crypto\" is not supported",
        "    at file:///var/task/index.mjs:1:383",
        "    at file:///var/task/index.mjs:2:15053",
        "    at file:///var/task/index.mjs:1:495",
        "    at file:///var/task/index.mjs:2:24330",
        "    at file:///var/task/index.mjs:1:495",
        "    at file:///var/task/index.mjs:81:38682",
        "    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)"
    ]
}
```

_Originally posted by @boringContributor in https://github.com/aws/aws-sam-cli/issues/4827#issuecomment-1688829726_

**Comment by hnnasit** [comment]
Hi @boringContributor, thanks for reporting the issue. Are you able to build your function without `sam build`? Starting from SAM CLI 1.85.0, the `banner` flag can be passed down to esbuild which seems to fix the require issue. This seems like a  known [issue](https://github.com/evanw/esbuild/issues/1921) and also mentions a similar [error](https://github.com/evanw/esbuild/issues/1921#issuecomment-1143530609) on the esbuild side. Could you check if the workarounds mentioned in the issue works for you?

**Comment by boringContributor** [comment]
@hnnasit thanks for the reply. The provided template.yml example I pasted on my original question adds this Banner though. I'm using a version greater than 1.85.0 and I still run into this error I described. The banner is added in my output file, but the deployed version has a runtime error with the described error.

I can build my function without `sam build` yes. Even `sam build` builds locally and the build output looks correct. Still throws the error at runtime.
